perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,475 @@
+        if from_date:
+            from_date_str = from_date.isoformat()
+        page_allrevs_query = "action=query&list=allrevisions"
+        page_allrevs_query += "&arvnamespace="+"|".join(namespaces)
+        page_allrevs_query += "&arvdir=newer&arvlimit=max&format=json"
+        # Minize the revision info. It is not used.
+        page_allrevs_query += "&arvprop=ids"
+        if arvcontinue:
+            page_allrevs_query += "&arvcontinue="+arvcontinue
+        else:
+            if from_date:
+                page_allrevs_query += "&arvstart="+from_date_str
+        allrevisions_url = self.api_url + "?" + page_allrevs_query
+        logger.debug("All revisions query: %s", (allrevisions_url))
+        page_revs = requests.get(allrevisions_url)
+        return page_revs.text
+
+class MediaWikiCommand(BackendCommand):
+    """Class to run MediaWiki backend from the command line."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.url = self.parsed_args.url
+        self.from_date = str_to_datetime(self.parsed_args.from_date)
+        self.origin = self.parsed_args.origin
+        self.outfile = self.parsed_args.outfile
+
+        if not self.parsed_args.no_cache:
+            if not self.parsed_args.cache_path:
+                base_path = os.path.expanduser('~/.perceval/cache/')
+            else:
+                base_path = self.parsed_args.cache_path
+
+            cache_path = os.path.join(base_path, self.url)
+
+            cache = Cache(cache_path)
+
+            if self.parsed_args.clean_cache:
+                cache.clean()
+            else:
+                cache.backup()
+        else:
+            cache = None
+
+        self.backend = MediaWiki(self.url, cache=cache, origin=self.origin)
+
+    def run(self):
+        """Fetch and print the Builds.
+
+        This method runs the backend to fetch the Builds of a given url.
+        Builds are converted to JSON objects and printed to the
+        defined output.
+        """
+        if self.parsed_args.fetch_cache:
+            pages = self.backend.fetch_from_cache()
+        else:
+            pages = self.backend.fetch(from_date=self.from_date)
+
+        try:
+            for build in pages:
+                obj = json.dumps(build, indent=4, sort_keys=True)
+                self.outfile.write(obj)
+                self.outfile.write('\n')
+        except requests.exceptions.HTTPError as e:
+            raise requests.exceptions.HTTPError(str(e.response.json()))
+        except IOError as e:
+            raise RuntimeError(str(e))
+        except Exception as e:
+            if self.backend.cache:
+                self.backend.cache.recover()
+            raise RuntimeError(str(e))
+
+    @classmethod
+    def create_argument_parser(cls):
+        """Returns the MediaWiki argument parser."""
+
+        parser = super().create_argument_parser()
+
+        # MediaWiki options
+        group = parser.add_argument_group('MediaWiki arguments')
```

Comment:
You can remove this because it is not used. There aren't any optional parameter related to this backend.